src/utility.ts

```diff
@@ -281,11 +298,18 @@ export class Utility {
             } else {
                 deviceType = "device";
             }
-            device.iconPath = context.asAbsolutePath(path.join("resources", `${deviceType}-${state}.svg`));
+            if (context) {
+                device.iconPath = context.asAbsolutePath(path.join("resources", `${deviceType}-${state}.svg`));
+            }
             return device;
         });
     }
 
+    public static async getNoneEdgeDeviceList(iotHubConnectionString: string): Promise<DeviceItem[]> {
+        const [deviceList, edgeDeviceIdSet] = await Promise.all([Utility.getIoTDeviceList(iotHubConnectionString), Utility.getEdgeDeviceIdSet(iotHubConnectionString)]);

```

Comment:
It is two API call, please refer to `getEdgeDeviceList` to improve the performance.

Reply:
fixed